Mathlib/Combinatorics/SimpleGraph/Acyclic.lean

```diff
@@ -197,4 +198,33 @@ lemma IsTree.card_edgeFinset [Fintype V] [Fintype G.edgeSet] (hG : G.IsTree) :
       refine (hG.existsUnique_path _ _).unique ((hf _).takeUntil _) ?_
       simp [h.ne]
 
+/-- A minimally connected graph is a tree. TODO : Iff version. -/
+lemma isTree_of_minimal_connected (h : Minimal Connected G) : IsTree G := by
+  rw [isTree_iff, and_iff_right h.prop, isAcyclic_iff_forall_adj_isBridge]
+  exact fun _ _ _ ↦ by_contra fun hbr ↦ h.not_prop_of_lt (by simpa [← edgeSet_ssubset_edgeSet])
+    <| h.prop.connected_delete_edge_of_not_isBridge hbr
+
+/-- Every connected graph has a spanning tree. TODO : remove the `Fintype V` hypothesis. -/
+lemma Connected.exists_isTree_le [Fintype V] (h : G.Connected) : ∃ T ≤ G, IsTree T := by
```

Comment:
Unless it is necessary to make the statement we should always prefer `Finite` over `Fintype`:
```suggestion
/-- Every connected graph has a spanning tree. TODO : remove the `Finite V` hypothesis. -/
lemma Connected.exists_isTree_le [Finite V] (h : G.Connected) : ∃ T ≤ G, IsTree T := by
```
(`Finite` tells us _that_ something is finite and is true / false whereas `Fintype` tells us _how_ something is finite. `Fintype` is thus data and opens us up to diamond issues since there are multiple ways for something to be finite.)

Reply:
I'm very happy to be asked to do this! I'm not a fan of `Fintype`.